src/expander_pipeline.jl

```diff
@@ -1,5 +1,24 @@
 # Defines node "expanders" that transform nodes from the parsed markdown files.
 
+# helper for "cleaning up" content of modules to enable garbage collection.
+# See also <https://github.com/JuliaDocs/Documenter.jl/issues/2640>.
+function clear_module!(M::Module)
+    for name in names(M, all = true)
+        if !isconst(M, name)
+            @eval M $name = $nothing
+        end
+    end
+    return
+end
+
+function clear_modules!(d::Dict{Symbol, Any})
+    for (k, v) in d
+        startswith(String(k), "__atexample__") || continue
+        v isa Module && clear_module!(v)
+    end
```

Comment:
Should we also immediately trigger the GC here, to ensure that things get deallocated?

```suggestion
    end
    GC.gc()
```

Reply:
Would be OK by me. But of course this may not collect everything, for that we'd have to call `GC.gc(true)`, but that can be costly. So for the test I'd anyway insert a call to `GC.gc(true)` before checking whether the finalizer did run.

Reply:
On second thought in the test the GC should always be able to reclaim the freshly created object, so let's try with only this `GC.gc()` call